src/ItemDetail.js

```diff
@@ -58,6 +68,31 @@ class ItemDetail extends React.Component {
     console.log('request clicked');
   }
 
+  getRequests(item_name){
+    // GET request to get all outstanding requests for this item by this user
+    var url;
+    console.log(item_name);
+    if (localStorage.isAdmin == "true") {
+      url = "https://asap-test.colab.duke.edu/api/request/?item__name="+item_name+"&status=outstanding";
+    } else {
+      url = "https://asap-test.colab.duke.edu/api/request/?item__name="+item_name+"&status=outstanding";
+    }
+    xhttp.open("GET", url, false);
+    xhttp.setRequestHeader("Content-Type", "application/json");
+    xhttp.setRequestHeader("Authorization", "Bearer " + localStorage.token);
+    xhttp.send();
```

Comment:
Add a if(xhttp.status===500||xhttp.status===401){
 to catch the unsupported stuff